Dockerfile

```diff
@@ -0,0 +1,64 @@
+###############
+# Build stage #
+###############
+
+FROM  elixir:1.5-alpine
+
+WORKDIR /build
+
+# We need hex and rebar build tools installed.
+RUN mix local.hex --force && mix local.rebar --force
+
+# Copy all the source files. We can't just use `COPY . .` because that would
+# also copy deps, _build, and various other things that would cause trouble
+# with the build. Also, `COPY dir .` will copy the contents of the dir instead
+# of the dir itself, so we have to manually copy each dir into its own
+# destination.
+COPY mix.* ./
+COPY config ./config
+COPY lib ./lib
+COPY test ./test
+COPY rel ./rel
+
+# Build the release to put in the next image.
+ENV MIX_ENV=prod
+RUN mix deps.get
+RUN mix compile
+RUN mix release --env=prod
+
+
+###############
+# Image stage #
+###############
+
+# We don't actually need Erlang/Elixir installed, because it's all included in
+# the release package. Thus, we start from the base alpine image.
+FROM alpine:3.7
+
+# We need bash for the generated scripts, tini for signal propagation, and
+# openssl for crypto.
+RUN apk add --no-cache bash tini openssl
```

Comment:
`libressl` should be a drop-in replacement for `openssl` and is preferred on Alpine 3.5+.

Is distillery making the bash-specific scripts?

Reply:
Thanks, I'll update that.

Yes, distillery (or something, it might be Erlang stuff) is building bash-specific scripts. We could probably post-process them or something, but it seems more sensible to just install bash.

Reply:
Nope, looks like Erlang depends specifically on `openssl`:
```
2018-01-08 11:24:41 Unable to load crypto library. Failed with error:~n"~p, ~s"~nOpenSSL might not be installed on this system.~n
        load_failed
        "Failed to load NIF library: 'Error loading shared library libcrypto.so.1.0.0: No such file or directory (needed by /app/lib/crypto-4.2/priv/lib/crypto.so)'"
```

I've tried replacing `openssl` with `libressl` in the build image too, with no success. There's probably a way to get around that, but for now I think it's best to just stick with openssl. We're not really using it for anything, since all our comms are plain HTTP.